addOns/spider/src/main/java/org/zaproxy/addon/spider/automation/SpiderJob.java

```diff
@@ -559,6 +560,14 @@ public void setParseGit(Boolean parseGit) {
             this.parseGit = parseGit;
         }
 
+        public Boolean getParseDsStore() {
```

Comment:
Same here for tests.

Reply:
Why doesn't this work? (I think this exposes a pre-existing issue, perhaps bigger than spider)

```java

    @Test
    void shouldVerifyDsStoreParameterWhenNotSet() {
        String yamlStr =
                "parameters:\n"
                        + "  parseGit: true\n";
        AutomationProgress progress = new AutomationProgress();
        Yaml yaml = new Yaml();
        Object data = yaml.load(yamlStr);

        SpiderJob job = new SpiderJob();

        job.setJobData(((LinkedHashMap<?, ?>) data));

        // When
        job.verifyParameters(progress);

        // Then
        assertThat(job.getParameters().getParseDsStore(), is(equalTo(false)));
    }
```

If our yaml files say, ex:
```yaml
      parseDsStore:                    # Bool: Whether the spider will parse .DS_Store files in order to find URLs, default: false
```

Then if it isn't defined shouldn't it default?

Reply:
That's the expected behaviour, if you don't define the field in the plan it will return null. What the null means depends on the context, in some cases it might mean true in other cases false (it's up to the job to interpret that).